Sources/Prometheus/MetricTypes/Counter.swift

```diff
@@ -35,14 +40,15 @@ public class Counter<NumType: Numeric, Labels: MetricLabels>: Metric, Prometheus
         self.initialValue = initialValue
         self.value = initialValue
         self.prometheus = p
+        self.lock = Lock()
     }
     
     /// Gets the metric string for this counter
     ///
     /// - Returns:
     ///     Newline seperated Prometheus formatted metric string
-    public func getMetric(_ done: @escaping (String) -> Void) {
-        prometheusQueue.async(flags: .barrier) {
+    public func getMetric() -> String {
```

Comment:
I'm somewhat worried about this being `getMetric`, can we call it some `renderMetric` perhaps?
Getting does not imply that we'll start marshalling everything into the "wire format" which happens to be the strings...

It gets esp. worrying once we hit APIs like these which create multi-line strings. Would the `render` prefix be okey here you think @MrLotU ? 

Maybe @weissi has another naming idea.

Reply:
Based on https://forums.swift.org/t/discussion-swift-prometheus-implementation/25544 seems like `collect()` would be a good name for those :) Would be good to change before merging I think, the "get" wording is quite weird and it is public API

Reply:
Updated it to `.collect()` :)